cl/assets/react/DisclosureViewer.tsx

```diff
@@ -0,0 +1,463 @@
+      </h3>
+      <p className="v-offset-above-1">
+        CourtListener is a project of{' '}
+        <a href="https://free.law" target="_blank" rel="noreferrer">
+          Free Law Project
+        </a>
+        , a federally-recognized 501(c)(3) non-profit. We rely on donations for our financial security.
+      </p>
+      <p>Please support our work with a donation.</p>
+      <p>
+        <a href="" className="btn btn-danger btn-lg btn-block">
+          Donate Now
+        </a>
+      </p>
+    </div>
+  );
+};
+
+const Thumb = (data: Data) => {
+  return (
+    <div className="v-offset-below-4">
+      <h3>
+        <span>
+          Download
+          <a href={`/api/rest/v3/financial-disclosures/?id=${data.id}`}>
+            <i className="fa fa-code gray pull-right" />
+          </a>
+        </span>
+      </h3>
+      <hr />
+      <a href={data.filepath}>
+        {data.thumbnail != null ? (
+          <img
+            src={data.thumbnail}
+            alt="Thumbnail of Judge Portrait"
+            height={'150'}
+            className="img-responsive thumbnail shadow img-thumbnail judge-pic"
+          />
+        ) : (
+          <div
+            height={'150'}
+            alt="Thumbnail of Future Judge Portrait"
+            className={'well img-responsive thumbnail shadow img-thumbnail judge-pic'}
+          />
+        )}
+      </a>
+    </div>
+  );
+};
+
+const Sidebar = (
+  data: Data,
+  is_admin: boolean,
+  judge: Row[],
+  fetchJudge: React.ChangeEventHandler<HTMLInputElement> | undefined,
+  visible,
+  setVisible
+) => {
+  return (
+    <div>
+      {is_admin ? AdminPanel(data) : ''}
+      {Thumb(data)}
+      {Notes()}
+      {SearchPanel(judge, fetchJudge, visible, setVisible)}
+      {Support()}
+    </div>
+  );
+};
+
+const Notes = () => {
+  return (
+    <div className={'v-offset-below-4'}>
+      <h3>
+        <span>
+          Notes <i className="fa fa-sticky-note-o pull-right" />
+        </span>
+      </h3>
+      <hr />
+      <span>The data in this file was extracted with OCR technology and may contain typos.</span>
+      <ul className={'v-offset-above-2 v-offset-below-2'}>
+        <li>
+          <i className="fa fa-file-text-o gray" /> Links to the PDF row (if possible).
+        </li>
+        <li>
+          <i className="fa fa-file-excel-o black" /> The row may contain a redaction
+        </li>
+        <li>
+          <i className="fa fa-eye-slash black" /> Indicates the OCR identified data in the row but could not extract it
+        </li>
+      </ul>
+      <span>
+        You can read more about financial disclosures at the{' '}
+        <a href={'https://www.uscourts.gov/sites/default/files/guide-vol02d.pdf'}>
+          Guide to Judiciary Policy on Ethics and Judicial Conduct
+        </a>
+        .
+      </span>
+    </div>
+  );
+};
+
+const SearchPanel = (judge: Row[], fetchJudge: React.ChangeEventHandler<HTMLInputElement> | undefined, visible, setVisible) => {
```

Comment:
Surprising amount of repetition here. I'd expect a reusable component with the exception of a few tweaks? Like, maybe something like `<JudgeDisclosureSearch small=true>`

Reply:
This is trouble otherwise. Already I see things here I mentioned elsewhere. You know this though, I'm going to skip reading the rest of this component.

Reply:
Yes - that would be preferable and I think I should be able to do that - but I didn't get to it today because - as you pointed out - it was a little different.  I can try to tackle that in the morning.